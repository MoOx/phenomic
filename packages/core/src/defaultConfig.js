import path from "path";

export default {
  path: process.cwd(),
  outdir: path.join(process.cwd(), "dist"),
  port: 3333,
  bundleName: "phenomic.main",
  plugins: []
};
