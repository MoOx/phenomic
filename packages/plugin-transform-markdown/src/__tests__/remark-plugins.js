import defaultConfig from "@phenomic/core/lib/defaultConfig.js";

import remarkPlugins from "../remark-plugins.js";

it("should render markdown as json (react component)", () => {
  const input = "## Test\n[link](href)\n```js\nconsole.log(window)\n```";
  const md = remarkPlugins(
    {
      ...defaultConfig,
      plugins: [{ name: "@phenomic/plugin-renderer-react" }]
    },
    input
  ).contents;
  expect(md).toMatchSnapshot();
});

it("should render markdown as html", () => {
  const input = "## Test\n[link](href)\n```js\nconsole.log(window)\n```";
  const md = remarkPlugins(defaultConfig, input).contents;
  expect(md).toMatchSnapshot();
});
