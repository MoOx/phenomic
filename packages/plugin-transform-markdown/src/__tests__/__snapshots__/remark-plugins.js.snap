// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render markdown as html 1`] = `
"<h2 id=\\"test\\"><a href=\\"#test\\" class=\\"phenomic-HeadingAnchor\\">#</a>Test</h2>
<p><a href=\\"href\\">link</a></p>
<pre><code class=\\"language-js\\">console.log(window)
</code></pre>
"
`;

exports[`should render markdown as json (react component) 1`] = `
Object {
  "c": Array [
    Object {
      "c": Array [
        Object {
          "c": Array [
            "#",
          ],
          "p": Object {
            "className": "phenomic-HeadingAnchor",
            "href": "#test",
          },
          "t": "a",
        },
        "Test",
      ],
      "p": Object {
        "id": "test",
      },
      "t": "h2",
    },
    "
",
    Object {
      "c": Array [
        Object {
          "c": Array [
            "link",
          ],
          "p": Object {
            "href": "href",
          },
          "t": "a",
        },
      ],
      "t": "p",
    },
    "
",
    Object {
      "c": Array [
        Object {
          "c": Array [
            "console.log(window)
",
          ],
          "p": Object {
            "className": "language-js",
          },
          "t": "code",
        },
      ],
      "t": "pre",
    },
  ],
  "t": "div",
}
`;
